tests/utils/test_config.py

```diff
@@ -93,3 +94,61 @@ def test_from_configuration_files_get_typed_value(tmp_path_factory: pytest.TempP
 
     assert Config().get_bool("legacy-current-snapshot-id")
     assert Config().get_int("max-workers") == 4
+
+
+@pytest.mark.parametrize(
+    "config_location, config_content, expected_result",
+    [
+        (
+            "config",
+            {"catalog": {"default": {"uri": "https://service.io/api"}}},
+            {"catalog": {"default": {"uri": "https://service.io/api"}}},
+        ),
+        (
+            "home",
+            {"catalog": {"default": {"uri": "https://service.io/api"}}},
+            {"catalog": {"default": {"uri": "https://service.io/api"}}},
+        ),
+        (
+            "current",
+            {"catalog": {"default": {"uri": "https://service.io/api"}}},
+            {"catalog": {"default": {"uri": "https://service.io/api"}}},
+        ),
+        ("none", None, None),
+    ],
+)
+def test_from_multiple_configuration_files(
+    monkeypatch: pytest.MonkeyPatch,
+    tmp_path_factory: pytest.TempPathFactory,
+    config_location: str,
+    config_content: Optional[Dict[str, Any]],
+    expected_result: Optional[Dict[str, Any]],
+) -> None:
+    def create_config_file(directory: str, content: Optional[Dict[str, Any]]) -> None:
+        config_file_path = os.path.join(directory, ".pyiceberg.yaml")
+        with open(config_file_path, "w", encoding="utf-8") as file:
```

Comment:
nit:
```suggestion
        with open(config_file_path, "w", encoding=UTF8) as file:
```

Reply:
This was adjusted in https://github.com/apache/iceberg-python/pull/1464/commits/fe029fe40340186363bd4031ebb07a2219336d72.